teensy4/usb_desc.h

```diff
@@ -994,6 +970,55 @@ let me know?  http://forum.pjrc.com/forums/4-Suggestions-amp-Bug-Reports
 
 #endif
 
+// Avoid the need to maintain audio-related macros for every build type:
+#if defined(AUDIO_INTERFACE) // signals one of the audio-capable builds is in use
+	#if defined(AUDIO_USB_CHANNEL_COUNT) // channel count defined on command line via Tools menu
+		#undef AUDIO_CHANNELS // ovveride value set in this file
+		#define AUDIO_CHANNELS AUDIO_USB_CHANNEL_COUNT
+	#endif // defined(AUDIO_USB_CHANNEL_COUNT)
+
+	// These macros should set the wMaxPacketSize and bInterval values so they're valid,
+	// in particular wMaxPacketSize <= 1024. However this seems to break the audio input
+	// code, so for the moment we go with invalid packet sizes and working audio...
+	//
+	// Change this value to 1024 for correct operation:
+	//                                      VVVVV
+	#define AUDIO_PACKET_SIZE(sz) (((sz) <= 91024)?(sz):((sz)/2))
+	#define AUDIO_INTERVAL(sz)    (((sz) <= 91024)?4:3)
```

Comment:
I think this operation might need to be rethought. Personally, here's what I was thinking, just based on my use:

* `AUDIO_INTERVAL` is a constant derived from `(AUDIO_FREQUENCY, AUDIO_CHANNELS, AUDIO_SAMPLE_BYTES)` by default, but is also override-able on the CLI just like you did to `AUDIO_CHANNELS` (for my case, for example, I'll be wanting the highest frequency), and then
* `AUDIO_PACKET_SIZE` is a constant derived from `(AUDIO_FREQUENCY, AUDIO_CHANNELS, AUDIO_SAMPLE_BYTES, AUDIO_INTERVAL)`

I'll play with it and try to make a version that looks like it to see how it goes. Let me know if you disagree with the approach.

Reply:
That sounds fair. The above was intended as a starting point for keeping the value of AUDIO_PACKET_SIZE() less than or equal to 1024, and only needed two choices for AUDIO_INTERVAL() to get to 96k/8ch/16bit. But a more flexible arrangement is always good...

I believe AUDIO_INTERVAL values have to be between 4 (for a 1ms update) and 1 (for 125µs). I can't see any particular merit in forcing it to a higher frequency than is necessary to get the packet size <=1024 bytes, but there's no harm in having the option. (Technically you can go to bInterval = 16 which is a 4.096s interval! See section 5.6.4 of the USB 2.0 specification).

I had issues with bInterval = 3 / wMaxPacketSize = 776 bytes, which should be about right for 96/8/16. I was getting bursts of USB interrupts at the expected 500µs intervals, but then it'd go quiet so I ended up with an average of an interrupt every 1ms. Possibly a Windows weirdness, maybe related to caching descriptors rather than using the ones just read from the device. You may get better results with Linux.